pcl_ros/src/pcl_ros/surface/moving_least_squares.cpp

```diff
@@ -212,7 +212,22 @@ pcl_ros::MovingLeastSquares::config_callback (MLSConfig &config, uint32_t /*leve
   {
     use_polynomial_fit_ = config.use_polynomial_fit;
     NODELET_DEBUG ("[config_callback] Setting the use_polynomial_fit flag to: %d.", use_polynomial_fit_);
+#if PCL_VERSION_COMPARE(<, 1, 9, 0)
     impl_.setPolynomialFit (use_polynomial_fit_);
+#else
+    if (use_polynomial_fit_)
+    {
+      NODELET_WARN ("[config_callback] use_polynomial_fit is deprecated, use polynomial_order instead!");
+      if (impl_.getPolynomialOrder () < 2)
```

Comment:
Rather than checking `< 2` why not use the `config.polynomial_order` already around?

I don't think the `else` needs to be there. Even the normal polynomial order API if the user asked for it didn't do that: https://github.com/ros-perception/perception_pcl/blob/acb450cc960694afc1e3d9cac525fc5063e71982/pcl_ros/src/pcl_ros/surface/moving_least_squares.cpp#L236